pkg/apis/core/types_shoot.go

```diff
@@ -1205,6 +1208,30 @@ type MaintenanceAutoUpdate struct {
 	MachineImageVersion *bool
 }
 
+// MaintenanceAutoRotation contains information about which rotations should be automatically performed.
+type MaintenanceAutoRotation struct {
+	// Credentials contains information about which credentials should be automatically rotated.
+	Credentials *MaintenanceCredentialsAutoRotation
+}
+
+// MaintenanceCredentialsAutoRotation contains information about which credentials should be automatically rotated.
+type MaintenanceCredentialsAutoRotation struct {
+	// ETCDEncryptionKey configured the automatic rotation for the etcd encryption key.
```

Comment:
```suggestion
	// ETCDEncryptionKey configures the automatic rotation for the etcd encryption key.
```
Also adapt the v1beta1 type.